src/NServiceBus.Persistence.AzureStorage/Subscriptions/AzureSubscriptionStorage.cs

```diff
@@ -142,5 +190,11 @@ public int GetHashCode(Subscriber obj)
                 return (obj.Endpoint ?? string.Empty).Length;
             }
         }
+
+        internal class CacheItem
+        {
+            public DateTime Stored;
+            public Task<IEnumerable<Subscriber>> Subscribers;
```

Comment:
>  Would storing awaited result be cheaper? 

You mean storing tasks. Yes, I think that would be effective. You don't need to store `Task.FromResult(await GetSub())`, just a task returned by the call. If task is awaited, then it's completed. And completed task`s result can be used multiple times.